x-pack/plugins/cases/server/authorization/utils.test.ts

```diff
@@ -0,0 +1,293 @@
+                Object {
+                  "type": "literal",
+                  "value": false,
+                },
+              ],
+              "function": "is",
+              "type": "function",
+            },
+          ],
+          "function": "and",
+          "type": "function",
+        }
+      `);
+    });
+  });
+
+  describe('includeFieldsRequiredForAuthentication', () => {
+    it('returns undefined when the fields parameter is not specified', () => {
+      expect(includeFieldsRequiredForAuthentication()).toBeUndefined();
+    });
+
+    it('returns an array with a single entry containing the owner field', () => {
+      expect(includeFieldsRequiredForAuthentication([])).toStrictEqual([OWNER_FIELD]);
+    });
+
+    it('returns an array without duplicates and including the owner field', () => {
+      expect(includeFieldsRequiredForAuthentication(['a', 'b', 'a'])).toStrictEqual([
+        'a',
+        'b',
+        OWNER_FIELD,
+      ]);
+    });
+  });
+
+  describe('ensureFieldIsSafeForQuery', () => {
+    it("throws an error if field contains character that aren't safe in a KQL query", () => {
+      expect(() => ensureFieldIsSafeForQuery('id', 'cases-*')).toThrowError(
+        `expected id not to include invalid character: *`
+      );
+
+      expect(() => ensureFieldIsSafeForQuery('id', '<=""')).toThrowError(
+        `expected id not to include invalid character: <=`
+      );
+
+      expect(() => ensureFieldIsSafeForQuery('id', '>=""')).toThrowError(
+        `expected id not to include invalid character: >=`
+      );
+
+      expect(() => ensureFieldIsSafeForQuery('id', '1 or caseid:123')).toThrowError(
+        `expected id not to include whitespace and invalid character: :`
+      );
+
+      expect(() => ensureFieldIsSafeForQuery('id', ') or caseid:123')).toThrowError(
+        `expected id not to include whitespace and invalid characters: ), :`
+      );
+
+      expect(() => ensureFieldIsSafeForQuery('id', 'some space')).toThrowError(
+        `expected id not to include whitespace`
+      );
+    });
+
+    it("doesn't throw an error if field is safe as part of a KQL query", () => {
+      expect(() => ensureFieldIsSafeForQuery('id', '123-0456-678')).not.toThrow();
+    });
+  });
+
+  describe('getOwnersFilter', () => {
+    it('constructs a KueryNode with only a single node', () => {
+      expect(getOwnersFilter('a', ['hello'])).toMatchInlineSnapshot(`
+        Object {
+          "arguments": Array [
+            Object {
+              "type": "literal",
+              "value": "a.attributes.owner",
+            },
+            Object {
+              "type": "literal",
+              "value": "hello",
+            },
+            Object {
+              "type": "literal",
+              "value": false,
+            },
+          ],
+          "function": "is",
+          "type": "function",
+        }
+      `);
+    });
+
+    it("constructs a KueryNode or'ing together two filters", () => {
```

Comment:
What will happen if we pass an empty array?

Reply:
Great point. We should probably return undefined 👍 . I'll add that and a test.